lib/web_ui/test/golden_tests/engine/canvas_draw_image_golden_test.dart

```diff
@@ -50,6 +50,7 @@ void main() async {
     RecordingCanvas(const Rect.fromLTRB(0, 0, 400, 300));
     rc.save();
     rc.drawImage(createTestImage(), Offset(0, 0), new Paint());
+    rc.restore();
```

Comment:
Restores are optional in Flutter AFAIK. What happens if you do not call it?

Reply:
Changed [_CanvasPool] to count save/restore to handle case.